src/psyclone/core/symbolic_maths.py

```diff
@@ -99,7 +99,11 @@ def equal(exp1, exp2):
         if exp1 is None or exp2 is None:
             return exp1 == exp2
 
-        return SymbolicMaths._subtract(exp1, exp2) == 0
+        diff = SymbolicMaths._subtract(exp1, exp2)
+        # Add support for ranges, in which case all values must be equal
+        if isinstance(diff, list):
+            return all(i == 0 for i in diff)
```

Comment:
That caused me some headache :) After some experimentation: in general '0' seems to be best, since sympy will compare this as expected, But as long as the output is a sympy expressions (which we always have after calling _substract), it does actually work as expected. So I changed it to numbers.Zero to be consistent.